### PR TITLE
MINIFICPP-590: Fix zero length files at startup

### PR DESCRIPTION
**Pull request by phrocker**
MINIFICPP-589: Institute regex filtering for interfaces

MINIFICPP-589: Limit Pcap to 4.9.4+

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by phrocker** [comment]
Did not experience issue noted in MINIFICPP-590. The only issue were zero length files at predictable times due to memory collisions. Batching appears to work as designed.

File: extensions/pcap/CapturePacket.cpp [hunk 1]
```diff
@@ -54,13 +54,15 @@ namespace processors {
 std::shared_ptr<utils::IdGenerator> CapturePacket::id_generator_ = utils::IdGenerator::getIdGenerator();
 core::Property CapturePacket::BaseDir("Base Directory", "Scratch directory for PCAP files", "/tmp/");
 core::Property CapturePacket::BatchSize("Batch Size", "The number of packets to combine within a given PCAP", "50");
+core::Property CapturePacket::NetworkControllers("Network Controllers", "List of network controllers to attach to -- each may be a regex", ".*");
```
[the remark this answers is not in the document]
Comment (phrocker) [reply extensions/pcap/CapturePacket.cpp]: This is meant to convey a YAML list ( https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html ), however as this isn't intended to be YAML specific I will clarify this. I thought about making it dynamic but did not see great value. Do you think this should be dynamic?

File: extensions/pcap/CapturePacket.cpp [hunk 2]
```diff
@@ -54,13 +54,15 @@ namespace processors {
...
 core::Property CapturePacket::CaptureBluetooth("Capture Bluetooth", "True indicates that we support bluetooth interfaces", "false");
 
 const char *CapturePacket::ProcessorName = "CapturePacket";
 
 std::string CapturePacket::generate_new_pcap(const std::string &base_path) {
   std::string path = base_path;
   // can use relaxed for a counter
+  //int cnt = num_.fetch_add(1, std::memory_order_relaxed);
```
[the remark this answers is not in the document]
Comment (phrocker) [reply extensions/pcap/CapturePacket.cpp]: err thanks can't believe I missed that.

File: extensions/pcap/CapturePacket.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (apiri) [reply extensions/pcap/CapturePacket.cpp]: I would agree that dynamic doesn't really seem like the right fit and arguably a misuse of what they are supposed to represent.  The clarification is definitely helpful though concerning the expectation of the YAML list.  Admittedly this comment stemmed from my NiFI leaning perspective where all property values are inherently just strings

**Comment by apiri** [comment]
As long as we have a ticket for it that works for me.  Will review the updates.

**Pull request closed without merging**